sim/druid/item_sets_pve_phase_6.go

```diff
@@ -183,15 +183,33 @@ var ItemSetGenesisFury = core.NewItemSet(core.ItemSet{
 
 // Each time you Dodge while in Dire Bear Form, you gain 10% increased damage on your next Mangle or Swipe, stacking up to 5 times.
 func (druid *Druid) applyTAQGuardian2PBonus() {
-	label := "S03 - Item - TAQ - Druid - Guardian 2P Bonus"
-	if druid.HasAura(label) {
-		return
-	}
+	druid.Tank2PieceAqProcAura = druid.RegisterAura(core.Aura{
+		Label:     "Guardian 2P Bonus Proc",
+		ActionID:  core.ActionID{SpellID: 1213188},
+		Duration:  time.Second * 10,
+		MaxStacks: 5,
+		OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks, newStacks int32) {
+			druid.MangleBear.DamageMultiplierAdditive += 0.1 * float64(newStacks-oldStacks)
+			druid.SwipeBear.DamageMultiplierAdditive += 0.1 * float64(newStacks-oldStacks)
+		},
+	})
 
-	druid.RegisterAura(core.Aura{
-		Label: label,
-		OnInit: func(aura *core.Aura, sim *core.Simulation) {
-			// TODO
+	druid.Tank2PieceAqAura = druid.RegisterAura(core.Aura{
+		Label:    "S03 - Item - TAQ - Druid - Guardian 2P Bonus",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitTaken: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if druid.form == Bear && spell.ProcMask.Matches(core.ProcMaskMelee) && result.Outcome.Matches(core.OutcomeDodge) {
+				druid.Tank2PieceAqProcAura.Activate(sim)
+				druid.Tank2PieceAqProcAura.AddStack(sim)
+			}
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell.SpellCode == SpellCode_DruidMangleBear || spell.SpellCode == SpellCode_DruidSwipeBear {
+				druid.Tank2PieceAqProcAura.SetStacks(sim, 0)
+			}
 		},
 	})
```

Comment:
Just simplifying with the `MakePermanent` helper

```suggestion
	core.MakePermanent(druid.Tank2PieceAqAura = druid.RegisterAura(core.Aura{
		Label:    "S03 - Item - TAQ - Druid - Guardian 2P Bonus",
		OnSpellHitTaken: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
			if druid.form == Bear && spell.ProcMask.Matches(core.ProcMaskMelee) && result.Outcome.Matches(core.OutcomeDodge) {
				druid.Tank2PieceAqProcAura.Activate(sim)
				druid.Tank2PieceAqProcAura.AddStack(sim)
			}
		},
		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
			if spell.SpellCode == SpellCode_DruidMangleBear || spell.SpellCode == SpellCode_DruidSwipeBear {
				druid.Tank2PieceAqProcAura.SetStacks(sim, 0)
			}
		},
	}))
```